source/common/upstream/http_status_checker.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <vector>
+
+#include "envoy/api/v2/core/health_check.pb.h"
+
+#include "common/protobuf/protobuf.h"
+
+namespace Envoy {
+namespace Upstream {
+
+class HttpStatusChecker;
```

Comment:
overall comment about this class: should we have this as a class in its own file, or given that the only use is in the http health checker, should we make it a struct within the http health checker for now?

Reply:
There are several places in the codebase were we follow that pattern for small abstractions that only relate to one class.

Reply:
I generally prefer smaller files with less responsibility, separate testing etc. so I made it this way.
As this is my first PR in this project I am not aware of general patterns you use. I can refactor this into inner class if you decide so.